sdk/keyvault/azure-security-keyvault-keys/test/ut/key_cryptographic_client_test_live.cpp

```diff
@@ -233,13 +233,13 @@ TEST_F(KeyVaultClientTest, RemoteSignVerifyRSA384)
   }
 }
 
-TEST_F(KeyVaultClientTest, RemoteSignVerifyDataRSA256)
+TEST_P(KeyVaultClientTest, RemoteSignVerifyDataRSA256)
 {
   KeyClient keyClient(m_keyVaultUrl, m_credential);
   std::string keyName(GetUniqueName());
 
   CreateRsaKeyOptions rsaKeyOptions(keyName);
-  rsaKeyOptions.KeySize = 2048;
+  rsaKeyOptions.KeySize = GetParam();
```

Comment:
This is neat, where is `GetParam()` defined? Does gtest auto-generate it based on `::testing::Values(13, 55, 233, 987, 1597, 2048),`?

Reply:
Get Param is part of gtest , it deals with reading the parameter generated out of that list 
https://github.com/google/googletest/blob/master/docs/advanced.md#value-parameterized-tests

Reply:
so the idea is you write one test , figure out want would be the variable part of the test, put the values in the list declared int he macro , and then the test runs for each parameter value/ element in the list, 
if there's need for multiple values for one test ( not the case here, but you can imagine file name and path, or an image width and height, etc.) you can pass in the list a struct, or tuple or some other blob, gtest doen't really care. 
When you do GetParam you get the same tuple or blob and then you unpack it in the test to get whatever data is relevant for the run.